src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,27 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        eventListeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        eventListeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        eventListeners.forEach(listener -> executor.submit(() -> listener.onEvent(event)));
     }
 
     public void shutdown() {
-
+        executor.shutdown();
```

Comment:
When shutting down the ExecutorService, it's important to also call 'awaitTermination' after 'shutdown' to ensure that all submitted tasks have completed execution. This helps prevent potential resource leaks.

Reply:
I believe that in this instance, it will overcomplicat the solution.

Reply:
In general, I agree - this is not necessary here.
But on the other hand we can have a little practice. What we want to do is to gracefully shutdown the executor, `shutdown()` method does not accept new tasks anymore which does the job eventually, but we can also add `executor.awaitTermination(timeout)` which will wait until all tasks complete and returns `false` if it did not happen during the `timeout` time, which we can use to shut it down forcefully in case if graceful takes too much time and we do not want to hold the resources for too long. But if we do this , we have to keep in mind that `shutdownNow()` will return the list of unfinished/not started tasks from it's queue, and we'd have to handle them somehow 
So here is what we can have:
```suggestion
        executor.shutdown();
        try {
            if (!executor.awaitTermination(60, TimeUnit.SECONDS)) {
                executor.shutdownNow();
            }
        } catch (InterruptedException ie) {
            executor.shutdownNow();
            Thread.currentThread().interrupt();
        }
```